extensions/libraries/redcore/api/api.php

```diff
@@ -199,9 +199,27 @@ public static function getPostedData()
 				}
 			}
 
-			// We try to transform it into XML
-			if (is_null($parsedData) && $xml = @simplexml_load_string($inputData))
+			$xml = @simplexml_load_string($inputData);
+
+			// We try to transform soap XML
+			if (is_null($parsedData) && $xml !== false)
 			{
+				/*
+					XML Namespaces suck and I couldn't get it to read the data accurately for SOAP
+					So I'm stripping everything between < and : in the xml and re-rendering it
+				    I know. I know. My skin is crawling too, but its the best I could do at the moment.
+				    If someone can make this work with namespaces that would be B-E-A-UTIFUL!
+				*/
+				$cleanData = preg_replace('/<(\/)?([a-z0-9]+):([a-z0-9]+)/i', '<$1$3', $inputData);
+				$xml = @simplexml_load_string($cleanData);
```

Comment:
I am not sure why you load xml string twice? What exact problem did you have with namespaces and loading of xml for SOAP. I am not even sure that you need this function for SOAP calls at all...
but this is not even the main problem here...
there is a reason why I put (is_null($parsedData) && $xml = @simplexml_load_string($inputData)) in the same bracket. It is a logic flow if that given string is not XML. Now it will try to convert it even if you provide a json type of string. This should be avoided....